Zimcheck does not handle internal links with local anchor properly

Checking a `physics.stackexchange.com_en_all_2020-04.zim`, it reports a lot of wrong fals positive bad internal links:
```
  A/question/531836.html (531836.html) was not found in article A/question/532236.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/532238.html
  I/question/\rm{state} (../../I/question/%5Crm%7Bstate%7D) was not found in article A/question/532839.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/532840.html
  A/question/ () was not found in article A/question/533607.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/533612.html
  A/question/ask.html (ask.html) was not found in article A/question/533849.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/533851.html
  A/question/451827/deep-concept-of-rotation-dynamics.html (../../question/451827/deep-concept-of-rotation-dynamics.html) was not found in article A/tag/angular-velocity/5.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/astrometrics/1.html
  A/question/451827/deep-concept-of-rotation-dynamics.html (../../question/451827/deep-concept-of-rotation-dynamics.html) was not found in article A/tag/reference-frames/23.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/reference-frames/24.html
  A/question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html (../../question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html) was not found in article A/tag/density/6.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/density/7.html
  A/question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html (../../question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html) was not found in article A/tag/pressure/22.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/pressure/23.html
  A/question/451827/deep-concept-of-rotation-dynamics.html (../../question/451827/deep-concept-of-rotation-dynamics.html) was not found in article A/tag/vectors/15.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/vectors/2.html
  A/question/451827/deep-concept-of-rotation-dynamics.html (../../question/451827/deep-concept-of-rotation-dynamics.html) was not found in article A/tag/rotational-dynamics/23.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/rotational-dynamics/24.html
  A/question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html (../../question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html) was not found in article A/tag/air/6.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/amplituhedron/1.html
  A/question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html (../../question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html) was not found in article A/tag/aircraft/2.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/aircraft/3.html
  A/question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html (../../question/528399/reason-for-lower-air-pressure-above-an-airplane-wing.html) was not found in article A/tag/gravity/34.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/tag/gravity/35.html
```

The reason is that `zimcheck` does not remove the local anchor before checking the link. This is wrong, it should.

@MiguelRocha I think the bug is still there but the ticket is incomplete. It is not only about local anchor (so something after `#`) but URL arguments (something after `?`). And the problem is still there, see with a smaller file:
```
$ zimcheck --url_internal conlang.stackexchange.com_en_all_2020-04.zim 
[INFO] Checking zim file conlang.stackexchange.com_en_all_2020-04.zim
[INFO] Verifying Articles' content.. 
[ERROR] Invalid external links found :
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/alltags.html
  A/element/59.html (../element/59.html#a59) was not found in article A/question/50.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/501.html
  A/user/109.html (../user/109.html) was not found in article A/question/66.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/670.html
  I/static/identicon/740.png (../../I/static/identicon/740.png) was not found in article A/question/721.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/744.html
  I/static/identicon/740.png (../../I/static/identicon/740.png) was not found in article A/question/748.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/75.html
  I/question/questions/tagged/diachronics (../../I/question/questions/tagged/diachronics) was not found in article A/question/1042.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/1049.html
  A/tag/tolkien-*.html (../tag/tolkien-%2A.html) was not found in article A/question/94.html
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/question/940.html
[INFO] Overall Test Status: Fail
[INFO] Total time taken by zimcheck: 1 seconds.
```

This is an example of error about URL arguments:
```
  -/static/MathJax.js?config=TeX-AMS_HTML-full (../-/static/MathJax.js?config=TeX-AMS_HTML-full) was not found in article A/alltags.html
```

and this is an example of error about local URL anchor:
```
  A/element/59.html (../element/59.html#a59) was not found in article A/question/50.html
```

@MiguelRocha Please open a separate ticket to check if the local anchor are valid (this is a really low prio).

--- a/src/zimcheck.cpp
+++ b/src/zimcheck.cpp
@@ -225,7 +225,7 @@ std::string normalize_link(const std::string& input, const std::string& baseUrl)
                 continue;
             }
         }
-        if ( *p == '#' )
+        if ( *p == '#' || *p == '?')
             // This is a beginning of the #anchor inside a page. No need to decode more
             break;
         if ( *p == '%')
@@ -400,7 +400,7 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
             auto links = getLinks(it->getData());
             for(auto olink: links)
             {
-                if (olink.front() == '#')
+                if (olink.front() == '#' || olink.front() == '?')
                     continue;
                 if (isInternalUrl(olink)) {
                     auto link = normalize_link(olink, baseUrl);

